projects/extension/ai/secrets.py

```diff
@@ -24,27 +24,21 @@ def remove_secret_from_cache(sd_cache: dict[str, str], secret_name: str):
 
 def get_secret(
     plpy,
-    secret: Optional[str],
-    secret_name: Optional[str],
-    secret_name_default: str,
-    sd_cache: Optional[dict[str, str]],
-) -> str:
+    secret: Optional[str] = None,
+    secret_name: Optional[str] = None,
+    secret_name_default: Optional[str] = None,
+    sd_cache: Optional[dict[str, str]] = None,
+) -> str | None:
     if secret is not None:
         return secret
 
     if secret_name is None:
         secret_name = secret_name_default
 
     if secret_name is None or secret_name == "":
-        plpy.error("secret_name is required")
-
-    secret = reveal_secret(plpy, secret_name, sd_cache)
-    if secret is None:
-        plpy.error(f"missing {secret_name} secret")
```

Comment:
What is the motivation for this change? If must avoid throwing an error, I think we should still emit a warning.

Reply:
We must avoid throwing an error. I'm okay with emitting something here, but I'm not convinced it should be a warning.

Two alternatives:
1. We could move the error to the caller, for callers who need the secret to be populated.
2. We could let the caller tell this function if it wants to error on missing secret or not (but tbh I would prefer option 1).

WDYT?

Reply:
I'm okay with option 1

Reply:
I ended up taking hidden option 3: don't modify anything in `secrets.py` and handle this in the caller of `ai.secrets.get_secret`.